app/models/document.rb

```diff
@@ -239,7 +239,7 @@ def publish!
         email_alert_api.send_alert(EmailAlertPresenter.new(self).to_json)
       end
 
-      publish_request.code == 200 && rummager_request.code == 200
+      publish_request.code == 200 && rummager_request.code == 202
```

Comment:
GDS Api Adapters does behave like this so the check does feel redundant.  The real question is that now Rummager returns 202 "Thanks, I'll get round to it later" what API does it give us for checking that there weren't errors with that?  We're accepting the 202 and telling users that the document is published, but something might happen with that eventual search processing that means it's not in the search index - how do we let users know about this?  The same question could be levelled at publishing-api too as although it responds with a 200, it's also really a 202 because everything is async now isn't it?  

Should publishing-api respond with a 202?  Should rummager respond with a 200? Should there be a "what's the state of my thing?" api for both?  Is there a pattern for this sort of thing?

Apols for hi-jacking this PR somewhat.